src/modules/content-sqlite/content-sqlite.c

```diff
@@ -374,20 +387,25 @@ void checkpoint_put_cb (flux_t *h,
 {
     struct content_sqlite *ctx = arg;
     const char *key;
-    const char *value;
+    json_t *o;
+    char *value = NULL;
 
     if (flux_request_unpack (msg,
                              NULL,
-                             "{s:s s:s}",
+                             "{s:s s:o}",
                              "key",
                              &key,
                              "value",
-                             &value) < 0)
+                             &o) < 0)
         goto error;
     if (strlen (key) == 0) {
         errno = EINVAL;
         goto error;
     }
+    if (!(value = json_dumps (o, JSON_ENCODE_ANY))) {
+        errno = EINVAL;
+        goto error;
```

Comment:
We really only need to support reading the version 0 checkpoint, not writing it, so JSON_ENCODE_ANY is not needed in the `json_dumps()` .  While you're there, may as well add JSON_COMPACT.  The strlen check can be dropped too.

Reply:
ahhh i remember why i did JSON_ENCODE_ANY before, it's b/c tons of tests like in `t0012-content-sqlite` still write general strings instead of objects.  Do we still want to support generically writing strings to checkpoint?  I assume no b/c this is specifically the kvs checkpoint put/get rpc.